Source/Core/VideoCommon/AVIDump.cpp

```diff
@@ -273,3 +287,15 @@ void AVIDump::DoState()
 {
   s_last_frame = CoreTiming::GetTicks();
 }
+
+void AVIDump::CheckResolution(int width, int height)
+{
+  if (width != s_current_width || height != s_current_height)
+  {
+    Stop(true);
```

Comment:
Since this is only called here with `true` as parameter, would it make sense to move the `s_file_index` change over here?